samples/apps/copilot-chat-app/webapi/Services/AppInsightsTelemetryService.cs

```diff
@@ -0,0 +1,95 @@
+/// </summary>
+public class AppInsightsTelemetryService : ITelemetryService
+{
+    private readonly TelemetryClient _telemetryClient;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    /// <summary>
+    /// Creates an instance of the app insights telemetry service.
+    /// This should be injected into the service collection during startup.
+    /// </summary>
+    /// <param name="telemetryClient">An AppInsights telemetry client</param>
+    /// <param name="httpContextAccessor">Accessor for the current request's http context</param>
+    public AppInsightsTelemetryService(TelemetryClient telemetryClient, IHttpContextAccessor httpContextAccessor)
+    {
+        this._telemetryClient = telemetryClient;
+        this._httpContextAccessor = httpContextAccessor;
+    }
+
+    /// <inheritdoc/>
+    public void TrackSkillEvent(string skillName, string functionName, bool success)
+    {
+        var properties = new Dictionary<string, string>(this.BuildDefaultProperties())
+        {
+            { "skillName", skillName },
+            { "functionName", functionName },
+            { "success", success.ToString() },
+        };
+
+        this._telemetryClient.TrackEvent("CompletionEvent", properties);
+    }
+
+    /// <summary>
+    /// Gets the current user's ID from the http context for the current request.
+    /// </summary>
+    /// <param name="contextAccessor">The http context accessor</param>
+    /// <returns></returns>
+    public static string? GetUserIdFromHttpContext(IHttpContextAccessor contextAccessor)
+    {
+        var context = contextAccessor.HttpContext;
+        if (context == null)
+        {
+            return GetUnknownUserId();
+        }
+
+        var user = context.User;
+        if (user == null || user.Identity == null || !user.Identity.IsAuthenticated)
+        {
+            return GetUnknownUserId();
+        }
+
+        var userId = user.FindFirst(ClaimTypes.NameIdentifier)?.Value;
+
+        if (userId == null)
+        {
+            return GetUnknownUserId();
+        }
+
+        return userId;
+    }
+
+    /// <summary>
+    /// Returns an identity for telemetry when the user is not authenticated.
+    /// </summary>
+    /// <returns>An id for an unauthenticated user</returns>
+    private static string GetUnknownUserId()
+    {
+        return $"unauthenticated";
+    }
+
+    /// <summary>
+    /// Prepares a list of common properties that all telemetry events should contain.
+    /// </summary>
+    /// <returns>Collection of common properties for all telemetry events</returns>
+    private Dictionary<string, string> BuildDefaultProperties()
+    {
+        string? userId = GetUserIdFromHttpContext(this._httpContextAccessor);
+        DateTime currentTime = DateTime.UtcNow;
+
+        return new Dictionary<string, string>
+        {
+            { "userId", userId ?? "unknown" },
+            { "timestamp", currentTime.ToString("O") }
```

Comment:
I believe this field is redundant as EventTelemetry objects already have a Timestamp field:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.applicationinsights.datacontracts.eventtelemetry?view=azure-dotnet